### ISSUE TITLE
Error message when navigating to google

### ISSUE THREAD

**Issue by KyleMit** [blunderboy/requestly#41]
Most pages load fine, but when loading plain old https://www.google.com/ I get following error screen:

> ### Aw, Snap
> Something went wrong while displaying this webpage.  To continue, reload or go to another page.
> If you're seeing this frequently, try these [suggestions](https://support.google.com/chrome/answer/95669)

![screenshot](http://i.imgur.com/iGjfBRH.png)

In disabling the extensions one by one, it looks like it works fine without requestly, but bombs consistently when it's enabled (even when I have no active rules).

I'm on Windows 7
I'm using Chrome 40.0.2214.115 m

Not sure what more debugging information you would need.

**Comment by blunderboy** [comment]
@KyleMit Thanks for opening issue. I am not yet sure what's causing this. Let me update my chrome to v40.0.* and I hope I could reproduce it.

**Comment by blunderboy** [comment]
@lazyvab I do not seem to have Windows7 machine. I upgraded my Chrome to v41.* on Ubuntu but it works fine for me.

@KyleMit Can you please send me your list of rules (although they are not active) still I would like to see if any of those rules hampering your page. If you do not wish to share your rules with me, you can try the following approach:

*Approach1*
1. Export all the rules to a file and save it on your machine for backup.
2. Delete all rules one by one and check if google.com opens fine.
3. If you are able to figure out which rule causes this, let me know the rule and then I can use the rule to fix the issue.
4. Delete your existing rules and import your rules from the file (step 1) again to get back the deleted ones.

*Approach2*
1. Enable Requestly
2. Open a new tab and press F12 to open Chrome console 
3. Click on console
4. In url bar open google.com

Let me know if you are able to see anything interesting in your console.

**Comment by KyleMit** [comment]
**Approach 1** - This still presents even when requestly is the only enabled extension and all the rules have been deleted

![Repo](http://i.imgur.com/4NzXvPb.gif)

**Approach 2** - Nothing interesting in the console except the standard navigation message.  This only shows up when you check to preserve the log,but it isn't put there by Requestly anyway.

![Console](http://i.imgur.com/9BTwhTK.png)

**Comment by KyleMit** [comment]
Appears to have been solved by

1. Exporting All Rules
2. Deleting All Rules
3. Restarting Browser
4. Importing All Rules

Now everything is loading fine will all rules and extensions enabled.

**Comment by blunderboy** [comment]
@KyleMit Quite Strange..I could not reproduce the issue. Just one more thing, Did you restart your browser after updating to v40.* ?

**Comment by lazyvab** [comment]
@KyleMit There is probably some cache issue after recent chrome update. Just clear your browser cache and it should work.

**Comment by blunderboy** [comment]
@lazyvab Thanks for finding the issue. Couple of other guys also pinged me about the same issue and clearing the cache solved the problem for them. Thanks a lot!!

@KyleMit Are we good to close this issue ? Please try clearing your cache if you face the problem and let me know if it works out for you.

I could not dive in deep to figure out the root cause but it seems that there is some extension api which might have changed or some bug fix in extension model which might have caused this.

**Comment by blunderboy** [comment]
Closing this issue. **Clearing your cache after upgrading to v40.* solved the issue**.

**Comment by KyleMit** [comment]
Yep, good to close.  Thanks for your help and a great tool

**Comment by sensboston** [comment]
You need to re-open the issue; the showstopper bug still exist and makes your extension very annoying... Clearing cache and other "tricks" are not helpful, it's just a temporary solutions. After few days this bug appear again.

**Comment by blunderboy** [comment]
@sensboston O man! This is really annoying to Users as well as the developer (Me). I am not really able to reproduce this issue on my machine (Chrome v41, Ubuntu 12.10). Which OS and Chrome Version are you on ? I will try to give my best support to help fix this.

**Comment by sensboston** [comment]
Windows 8.1, Chrome 42.0.2311.135 m, exts: AdBlock, Awesome Screenshot, Chrome apps & dev tool, Download Vimewo videos, Facebook Video Downloader, Hangouts, LastPass: Free Password Manager, Speed Dial 2

**Comment by sensboston** [comment]
BTW, I installed "Switcheroo Redirector" and (at least for a few hours) runs well doing same job (I need to replace some scripts in the http response to my own).

P.S. Have you ever thought about adding regex power to change a response body? Or could you recommend some extensions for this? I just need to change a couple variables in the script but now I'm redirecting hosts scripts to my modified copies of these scripts.

**Comment by blunderboy** [comment]
@sensboston Thanks for the information. I would request you **not** to uninstall the extension. Just keep it disabled until I get a reproducible machine. If I come to know of a solution I will inform you.

About modifying something in Response, Chrome does not allow it So I don't think there is any extension which fits this use case

**Comment by blunderboy** [comment]
@sensboston I installed Windows 8 in VMWare and setup Chrome 42 on it. I then installed Requestly on Chrome and installed the list of extensions on your machine but unfortunately I did not encounter this issue.. Damn!!

Could you please help me in debugging it by providing me the following information.

> Just make sure you export all your rules so that you don't lose your data.

1. If you open Amazon (or any other website) directly into the browser, does your tab Crash ?
Or is it just the google search result which is causing the page to crash.
2. Try deleting all the rules and then clear your cache. See if the issue still persists. 
3. Restart your browser.
4. Uninstall and Reinstall the Extension.

Last time, Clearing Cache solved the issue for everybody. That again I could not confirm because I am yet unable to reproduce this showstopper thing.

@KyleMit May I ask you if you are still facing the same issue or you are good using it.

**Comment by sensboston** [comment]
Hmm, I'm very sorry man but Chrome is my primary browser but your extension makes it very unstable (it's 101% your bug, not a Google). So, I uninstalled your stuff and now I'm happy with Switcheroo Redirector - it do the job without any issues.

P.S. I'm developer and don't have time to help you to test and resolve your bugs; I just let you know - you had a SERIOUS SHOWSTOPPER. Try to fix it yourself - nobody else interested ('cause of working alternatives in the Google store). Or you'll get a lot of "1 star" (in case you'll get enough downloads first).

**Comment by blunderboy** [comment]
After doing some research on Google, found another thread [0]. The very first comment is

> #1 hqm.andr...@gmail.com
I'm experiencing the same thing. Thanks to China's wall slowing things down, I've narrowed down the culprit to:

https://www.google.com/xjs/_/js/sdch=d/k=xjs.s.en_US.87HfdAGnAmg.O/m=sx,c,sb,cdos,cr,elog,jsa,r,hsm,j,p,d,csi/am=pIj6W4wwlAV1AlEH/rt=j/d=1/t=zcms/rs=ACT90oH8a6O_t9ht7smzQbw0is0Tv6IBtQ

The weird strings do not seem to be cookie dependent (confirmed using wget)

With ScriptSafe enabled, Chrome tends to generate a "decode failure", which seems to be extracting malformed gzip and generates a gigantic mess (if I interrupt it before it "finishes loading", sometimes I see a large mess of bad unicode characters). With ScriptSafe disabled, the script decodes correctly.

Looks like this is the same issue with Windows Chrome but I am just wondering what's really happening with only some of the Users.

[0]: http://code.google.com/p/scriptno/issues/detail?id=291

**Comment by sensboston** [comment]
blunderboy, currently my area of interests and development is on the Windows Phone/Windows Store apps, so I can't switch to debug your stuff. I'm in US, on Comcast cable, not behind a Chinese "great firewall" so I don't really care what's happened with some setups etc. 

What I can recommend to you (I know you've never asked for but...) - install Win 8.1 to virtual PC, install same extensions and try to get your bug appear, debug and fix it! And (it's a DAMN important!) NEVER CLOSE your critical bug after pseudo-"fixes" like "clear cache" or "clear history" or "knock on the tambourine". If you're an engineer, you shouldn't believe in bullshit but MUST use a debugger/traces/breakpoint :)

**Comment by blunderboy** [comment]
@sensboston No worries man!! Thanks for all the help you did. I seeked your help because it is reproducible on your machine and not on others'. 

Yeah, great Suggestion. I will keep this in mind and won't close any critical stuff without proper reasons. I have already prepared a setup within VmWare now and will work towards fixing it.

Thanks for the great help!!

**Comment by sensboston** [comment]
Good. Please let me know if you'll get progress on the issue, maybe I'll reinstall your extension again (and if you'll add some "on-the-fly hacks" as I've described in the few posts above - instead of redirecting whole script I just need to change a downloaded script body for a little) - with that abilty your extension will be great add-on to Chrome.

**Comment by DenisGorbachev** [comment]
@blunderboy I'm having the same issue as well on Ubuntu 14.04, Chrome 42.0.2311.135 (64-bit)

Could you please fix it? Like, please-please? :)

**Comment by blunderboy** [comment]
@DenisGorbachev This task is on high priority for me now. I am onto it and trying to figure out a solution soon. I am just wishing myself a good luck to reproduce it on my machine.

**Comment by blunderboy** [comment]
@DenisGorbachev Can you launch your Chrome Browser with the following command

> google-chrome --enable-logging=stder

Since you are on ubuntu, you can see the logs in Terminal. And now try to search something in Google. Let me know if you find something interesting in logs.

**Comment by DenisGorbachev** [comment]
@blunderboy Wow, the issue disappeared after I tried the following instructions:

1. Exporting All Rules
1. Deleting All Rules
1. Restarting Browser
1. Importing All Rules

I've tried to reproduce it, but couldn't as well. So I guess it works fine now...

Thank you very much for prompt reply! If the error comes back, I'll definitely enable logging and try to reproduce it.

**Comment by DenisGorbachev** [comment]
@blunderboy the bug manifested again! This time, I was armed with ``stderr``.

```
[13753:13788:0515/203052:WARNING:spdy_session.cc(2373)] Received HEADERS for invalid stream 7
[13753:13788:0515/203052:WARNING:spdy_session.cc(2373)] Received HEADERS for invalid stream 11
[133,3746404608:17:30:52.858104] Native Client module will be loaded at base address 0x0000702d00000000
[13753:13788:0515/203058:WARNING:spdy_session.cc(2373)] Received HEADERS for invalid stream 17
[13753:13788:0515/203210:WARNING:spdy_session.cc(2373)] Received HEADERS for invalid stream 55
[13791:13791:0515/173212:WARNING:ipc_message_attachment_set.cc(37)] MessageAttachmentSet destroyed with unconsumed descriptors: 0/1
```

Additional details:

1. These logs have resulted from multiple "Aw Snaps" in a succession (some Ctrl+R reloads, some brand new omnibox searches).
1. Some "Aw Snap" results didn't produce any error logs at all.
1. The "spdy_session.cc / Received HEADERS for invalid stream" error has manifested very frequently. Does Requestly have an issue with SPDY? :)
1. The bug resolved itself after browser restart (no cache clearing was necessary).

**Comment by blunderboy** [comment]
@DenisGorbachev Thanks a lot for the logs. I do not seem to find anything interesting here. I will install Ubuntu 14 with Chrome 42 in a VM and try to reproduce it. 

I have made couple of posts on Stackoverflow and [Google Chrome Extension] Forum to get some help.
Let me see if somebody gives me a gentle word of hint there.

[Stackoverflow]: http://stackoverflow.com/questions/30270968/google-omnibox-searches-getting-crashed-when-chrome-extension-is-installed-happ

[Google Chrome Extension]: https://groups.google.com/a/chromium.org/forum/#!topic/chromium-extensions/y25kJWgM_9M

**Comment by DenisGorbachev** [comment]
Sachin, that's a good idea. By the way, it's necessary to browse for a
decent period of time until the bug manifests.

**Comment by blunderboy** [comment]
As Suggested by @Rob--W on [Stackoverflow](http://stackoverflow.com/questions/30270968/google-omnibox-searches-getting-crashed-when-chrome-extension-is-installed-happ?noredirect=1#comment48655237_30270968) not to modify headers unconditionally which should reduce the impact of this bug. I opened Issue #51 and finished the task. This update will be available in Chrome Store as Requestly v3.3.4.

Posts on Stackoverflow and Google Forum assured that this is happening due to blocking response headers.

**Comment by DenisGorbachev** [comment]
@blunderboy Thanks! Let's see if it solves the issue.

**Comment by blunderboy** [comment]
@DenisGorbachev Requestly v3.3.4 is available on Chrome Store now. You must have got an update already. Hopefully, it should resolve the issue. Fingers Crossed ;)

**Comment by blunderboy** [comment]
@all I haven't heard of this issue striking back since the fix which went in release v3.3.4. Closing this down.

**Comment by Rob--W** [comment]
To see whether the issue is really fixed, could any of you who experienced the issue before add Google to your rules, and see whether Chrome crashes after some period of use?
